sdk/storage/azure-storage-blob/azure/storage/blob/aio/download_async.py

```diff
@@ -229,137 +130,241 @@ def __aiter__(self):
+    """
+
+    def __init__(
+            self,
+            clients=None,
+            config=None,
+            start_range=None,
+            end_range=None,
+            validate_content=None,
+            encryption_options=None,
+            max_concurrency=1,
+            name=None,
+            container=None,
+            encoding=None,
+            **kwargs
+    ):
+        self.name = name
+        self.container = container
+        self.properties = None
+        self.size = None
+
+        self._clients = clients
+        self._config = config
+        self._start_range = start_range
+        self._end_range = end_range
+        self._max_concurrency = max_concurrency
+        self._encoding = encoding
+        self._validate_content = validate_content
+        self._encryption_options = encryption_options or {}
+        self._request_options = kwargs
+        self._location_mode = None
+        self._download_complete = False
+        self._current_content = None
+        self._file_size = None
+        self._non_empty_ranges = None
+        self._response = None
+
+        # The service only provides transactional MD5s for chunks under 4MB.
+        # If validate_content is on, get only self.MAX_CHUNK_GET_SIZE for the first
+        # chunk so a transactional MD5 can be retrieved.
+        self._first_get_size = self._config.max_single_get_size if not self._validate_content \
+            else self._config.max_chunk_get_size
+        initial_request_start = self._start_range if self._start_range is not None else 0
+        if self._end_range is not None and self._end_range - self._start_range < self._first_get_size:
+            initial_request_end = self._end_range
+        else:
+            initial_request_end = initial_request_start + self._first_get_size - 1
+
+        self._initial_range, self._initial_offset = process_range_and_offset(
+            initial_request_start, initial_request_end, self._end_range, self._encryption_options
+        )
+
+    def __len__(self):
+        return self.size
+
+    async def _setup(self):
+        self._response = await self._initial_request()
+        self.properties = self._response.properties
+        self.properties.name = self.name
+        self.properties.container = self.container
 
         # Set the content length to the download size instead of the size of
         # the last range
-        self.properties.size = self.download_size
+        self.properties.size = self.size
 
         # Overwrite the content range to the user requested range
-        self.properties.content_range = 'bytes {0}-{1}/{2}'.format(self.offset, self.length, self.file_size)
-
-        # Set additional properties according to download type
-        if extra_properties:
-            for prop, value in extra_properties.items():
-                setattr(self.properties, prop, value)
+        self.properties.content_range = 'bytes {0}-{1}/{2}'.format(
+            self._start_range,
```

Comment:
Do we want to rename these too to offset, length...since we are exposing this to public,

Reply:
No - that's what they were called and I just changed them back. By the time the values reach the StreamDownloader code they have already been converted to range values and that's how they're handled and used here, so they should be named accordingly.

Reply:
That, and these values are all underscored and not part of the public API.